Source/WebCore/html/HTMLScriptElement.cpp

```diff
@@ -87,9 +88,68 @@ void HTMLScriptElement::didFinishInsertingNode()
 }
 
 // https://html.spec.whatwg.org/multipage/scripting.html#dom-script-text
-void HTMLScriptElement::setText(String&& value)
+ExceptionOr<void> HTMLScriptElement::setText(std::variant<RefPtr<TrustedScript>, String> value)
 {
-    setTextContent(WTFMove(value));
+    ExceptionOr<String> stringValueHolder = WTF::switchOn(
+        value,
+        [this](const String& str) -> ExceptionOr<String> {
+            return trustedTypeCompliantString(TrustedType::TrustedScript, *scriptExecutionContext(), str, "HTMLScriptElement text"_s);
+        },
+        [](const RefPtr<TrustedScript>& trustedScript) -> ExceptionOr<String> {
+            return trustedScript->toString();
+        }
+    );
+    if (stringValueHolder.hasException())
+        return stringValueHolder.releaseException();
+
+    setTextContent(stringValueHolder.releaseReturnValue());
+    return { };
+}
+
+String HTMLScriptElement::textContentForBindings() const
+{
+    return Node::textContent();
+}
+
+ExceptionOr<void> HTMLScriptElement::setTextContentForBindings(std::optional<std::variant<RefPtr<TrustedScript>, String>> value)
+{
+    ExceptionOr<String> stringValueHolder = WTF::switchOn(
+        value.value_or(emptyString()),
```

Comment:
Does this do the right thing for a null? Does `trustedTypeCompliantString` handle empty strings in the best possible way? Do we really expect the same behavior for a null as for an empty string? Is there test coverage for this?

Reply:
The final subtest of https://wpt.live/trusted-types/block-string-assignment-to-HTMLElement-generic.html tests assigning null to script.textContent, converting this null to an empty string for the enforcement code is also what Chrome does.